### PR TITLE
Passcode (CTF Flag) Function

### PR DESCRIPTION
**Pull request by bjrjk**

### CODE REVIEW

File: vj4/server.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from vj4 import app
 from vj4.util import options
 
-options.define('listen', default='http://127.0.0.1:8888', help='Server listening address.')
+options.define('listen', default='http://0.0.0.0:8888', help='Server listening address.')
```
Comment (moesoha) [inline vj4/server.py]: revert this line plz, you can use `--listen` to specify the address you want to listen on

File: vj4/ui/templates/partials/problem_sidebar_normal.html [hunk 2]
```diff
@@ -106,6 +106,10 @@ <h1 class="section__title">{{ _('Information') }}</h1>
         {{ _('(None)') }}
       {% endif %}
       </dd>
+      {% if pdoc['psdoc']['status']==1 and pdoc['flag'] %}
```
Comment (moesoha) [inline vj4/ui/templates/partials/problem_sidebar_normal.html]: spaces around operators

File: vj4/locale/zh_CN.yaml [hunk 3]
```diff
@@ -590,6 +590,7 @@ Created By: 创建人
 Judged By: 评测机
 Categories: 分类
 Tags: 标签
+Flag: Flag (通过码)
```
Comment (moesoha) [inline vj4/locale/zh_CN.yaml]: What a long name!

File: vj4/ui/templates/partials/problem_sidebar_normal.html (hunk 2 above)
Comment (moesoha) [reply vj4/ui/templates/partials/problem_sidebar_normal.html]: `vj4.constant.record.STATUS_ACCEPTED`

Comment (moesoha) [reply vj4/ui/templates/partials/problem_sidebar_normal.html]: and put `pdoc['flag']` in front of status?

File: vj4/handler/problem.py [hunk 4]
```diff
@@ -707,7 +707,8 @@ def split_tags(self, s):
   @base.sanitize
   async def post(self, *, pid: document.convert_doc_id, hidden: bool=False,
                  category: str, tag: str,
-                 difficulty_setting: int, difficulty_admin: str=''):
+                 difficulty_setting: int, difficulty_admin: str='',
+                 accode: str):
```
Comment (twd2) [inline vj4/handler/problem.py]: `=''`

File: vj4/model/adaptor/problem.py [hunk 5]
```diff
@@ -58,7 +58,8 @@ def get_categories():
   pid = await add(domain_id=dest_domain_id, owner_uid=owner_uid,
                   title=pdoc['title'], content=pdoc['content'],
                   pid=pid, hidden=hidden, category=pdoc['category'],
-                  data=data, tag=pdoc.get('tag', []))
+                  data=data, tag=pdoc.get('tag', []),
+                  accode=pdoc['accode'])
```
Comment (twd2) [inline vj4/model/adaptor/problem.py]: `.get('accode', '')`

File: vj4/ui/templates/problem_settings.html [hunk 6]
```diff
@@ -23,6 +23,7 @@ <h1 class="section__title">{{ _('Settings') }}</h1>
           {{ form.form_checkbox(label='Hidden', help_text='Note: Problem title may not be hidden.', name='hidden', value=pdoc['hidden']) }}
           {{ form.form_text(columns=12, label='Category', help_text='Format: category 1, sub category 1.1, sub category 1.2, ..., sub category 1.x, ..., category n, sub category n.1, sub category n.2, ..., sub category n.m, ...', name='category', value=pdoc['category']|join(', ')) }}
           {{ form.form_text(columns=12, label='Tags', help_text='Splitted by \', \'.', name='tag', value=pdoc['tag']|join(', ')) }}
+          {{ form.form_text(columns=12, label='ACCode', help_text='', name='accode', value=(pdoc['accode'] or '' )) }}
```
Comment (twd2) [inline vj4/ui/templates/problem_settings.html]: `help_text='This message will be presented to those whose submissions are accepted.'` and the translation.

File: vj4/locale/zh_CN.yaml [hunk 7]
```diff
@@ -790,4 +791,5 @@ Copy Problem To: 将题目复制到
...
\ No newline at end of file
+Instead of copying the test data directly, the test data of the copied problems will be linked to the test data of the source problems (called the source test data). Thus, the copied problems can observe the changes in the source test data. The permissions of the test data of the copied problems follow the source test data, e.g., you still might not download them but the judges can. By uploading some new test data, the link will be broken and the new test data will be used.: 虽然测试数据不会被复制，但是系统会将题目的测试数据链接到原题。因此在原题的数据被改动的时候，复制后的题目的数据也会同时改动。链接后的测试数据的权限以原题为准，比如您可能依旧无法下载数据，但是评测机可以。在题目设置页面中可以通过上传新的测试数据的方式，这个数据链接会被删除，以后将使用您的新测试数据。
+This message will be presented to those whose submissions are accepted.: 这条消息将会被展示给题目通过的做题者。
```
Comment (twd2) [inline vj4/locale/zh_CN.yaml]: "通过此题的用户"?

File: vj4/locale/zh_CN.yaml [hunk 8]
```diff
@@ -590,6 +590,7 @@ Created By: 创建人
 Judged By: 评测机
 Categories: 分类
 Tags: 标签
+ACCode: Flag 通过码
```
Comment (twd2) [inline vj4/locale/zh_CN.yaml]: `ac_msg`
`AC Message: 通过消息`

**Pull request merged**